### PR TITLE
completed ticket 9

### PR DESCRIPTION
**Pull request by Amir2341**

### CODE REVIEW

File: models/model.js [hunk 1]
```diff
@@ -52,3 +52,18 @@ exports.selectAllArticles = async () => {
...
+exports.selectCommentsById = async (id) => {
+  const { rows: comments } = await db.query(
+    "SELECT * FROM comments WHERE article_id = $1",
+    [id]
+  );
+  if (comments.length === 0) {
+    return Promise.reject({
+      status: 404,
+      msg: `No user found for article_id: ${id}`,
+    });
```
Comment (cjplatten) [inline models/model.js]: I'm not sure what this error is finding. If the article id exists but there are no comments associated in the databaase, has an error occured?

Comment (cjplatten) [reply models/model.js]: It also doesn't look like this was tested.

**Review by cjplatten — CHANGES_REQUESTED** [review]
Please see comments

File: models/model.js (hunk 1 above)
Comment (cjplatten) [reply models/model.js]: I see what this error is for now. I would argue that you're trying to combine two possible outcomes into one test - one of which is an error and the other which isn't. What happens if the id does exist but has no comments vs what is the id doesn't exist?

**Review by cjplatten — CHANGES_REQUESTED** [review]
Please see comment

**Comment by corkesw** [comment]
I see the logic for handling the non-existent id... would also be good to see a test for an article with no comment and assert what you would like the api to send back in this case. eg, we'd still get a key of comments but it's value would be an empty array

File: __tests__/app.test.js [hunk 2]
```diff
@@ -216,3 +216,49 @@ describe("GET /api/articles", () => {
...
+      .then(({ body }) => {
+        expect(body.msg).toBe("Resource not found");
+      });
+  });
+  test("status 200 for article exists but no comment", () => {
+    return request(app)
+      .get("/api/articles/7/comments")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body).toEqual({ comments: [] });
```
Comment (karlaferry) [inline __tests__/app.test.js:261]: I would test the `body.comments.length` here instead of testing this way.

**Review by karlaferry — APPROVED** [review]

**Pull request merged**